### PR TITLE
Characters are non negative

### PR DESCRIPTION
**Pull request by NhatDinh**
Please review and let me know if any changes are needed.
@kelloggm @panacekcz

### CODE REVIEW

**Comment by NhatDinh** [comment]
@panacekcz Should I clone, build, add SuppressWarnings in `UtilMDE.java` in `plume-lib` and make a PR to `https://github.com/mernst/plume-lib:master`?

File: checker/tests/index/NonnegativeChar.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class NonnegativeChar {
+    void foreach(char[] array) {
+        for (char value : array) ;
+    }
+
+    char constant() {
+        return Character.MAX_VALUE;
+    }
+
+    @SuppressWarnings("cast.unsafe")
```
[the remark this answers is not in the document]
Comment (kelloggm) [reply checker/tests/index/NonnegativeChar.java]: An example of similar (but more complicated) piece of code is in `ValueAnnotatedTypeFactory.ValueTreeAnnotator#visitTypeCast`

[the remark this answers is not in the document]
Comment (NhatDinh) [reply checker/tests/index/NonnegativeChar.java]: @kelloggm 
I could not find `LowerBoundTreeAnnotator`. So I wonder by `LowerBoundTreeAnnotator` did you mean `LowerBoundVisitor`?

**Comment by NhatDinh** [comment]
I've just commit the `visitTypeCast()` method to this PR
@kelloggm

**Comment by kelloggm** [comment]
Closing until I have time to work on this again.

**Pull request closed without merging**